tests/unit/test_placeholders.py

```diff
@@ -178,4 +178,170 @@ def check_immutable(placeholder):
             for k, v in placeholder.store.items():
                 return check_immutable(v)
     else:
-        return False
\ No newline at end of file
+        return False
+
+
+def test_map_item_value_creation_with_subscript_operator():
```

Comment:
Tests will be parametrized once [ResultSelector PR](https://github.com/aws/aws-step-functions-data-science-sdk-python/pull/102) is merged